sda/internal/database/db_functions.go

```diff
@@ -100,6 +100,45 @@ func (dbs *SDAdb) getInboxFilePathFromID(submissionUser, fileID string) (string,
 	return filePath, nil
 }
 
+// GetFileIDByUserPathAndStatus checks if a file exists in the database for a given user and submission filepath
+// and returns its fileID for the latest specified status
+func (dbs *SDAdb) GetFileIDByUserPathAndStatus(submissionUser, filePath, status string) (string, error) {
+	var (
+		err    error
+		fileID string
+	)
+	// 2, 4, 8, 16, 32 seconds between each retry event.
+	for count := 1; count <= RetryTimes; count++ {
+		fileID, err = dbs.getFileIDByUserPathAndStatus(submissionUser, filePath, status)
+		if err == nil || strings.Contains(err.Error(), "sql: no rows in result set") {
+			break
```

Comment:
When `err == nil`, the actual error caused by `sql: no rows in result set` will not be returned. 
Maybe change to 
```go
return "", sql.ErrNoRows
```

Reply:
Or Maybe 

```go
	if errors.Is(err, sql.ErrNoRows) {
		return "", sql.ErrNoRows
	}
```

will work. Not tested yet.

Reply:
The error is captured and returned correctly, see [here](https://github.com/neicnordic/sensitive-data-archive/blob/8d0998f20b55035cd166543a6707a77288cf54fc/sda/internal/database/db_functions_test.go#L1243) and the line below for the corresponding test.